oidc.go

```diff
@@ -126,6 +126,7 @@ var oidcCallbackTemplate = template.Must(
 	</html>`),
 )
 
+// TODO: Why is the entire machine registration logic duplicated here?
```

Comment:
I think there is fair amount of things that could be tried up about it yes, could you create an issue about it so we can try to track it?

Reply:
Done, #294